.drone.yml

```diff
@@ -118,23 +158,153 @@ type: kubernetes
-    api_server: https://api.prod.corp.mongodb.com
-    kubernetes_token:
-      from_secret: prod_kubernetes_token
-    values_files: [ "environments/prod.yaml" ]
+  - name: deploy-prod
+    image: quay.io/mongodb/drone-helm:v3
+    settings:
+      chart: mongodb/web-app
+      chart_version: 4.7.3
+      add_repos: [mongodb=https://10gen.github.io/helm-charts]
+      namespace: devrel
+      release: devcenter-frontend
+      values: image.tag=git-${DRONE_COMMIT_SHA:0:7},image.repository=795250896452.dkr.ecr.us-east-1.amazonaws.com/devrel/${DRONE_REPO_NAME},ingress.enabled=true,ingress.hosts[0]=devcenter-frontend.devrel.prod.corp.mongodb.com
+      api_server: https://api.prod.corp.mongodb.com
+      kubernetes_token:
+        from_secret: prod_kubernetes_token
+      values_files: ["environments/prod.yaml"]
+
+  - name: set-release-sha
+    image: drone/cli:1.4.0-alpine
+    commands:
+    # See comment to similar line above regarding usage.
+    - drone secret update --name app_release_production --data ${DRONE_COMMIT_SHA} mongodb/devcenter
+    environment:
+      DRONE_SERVER: ${DRONE_SYSTEM_PROTO}://${DRONE_SYSTEM_HOST}
+      DRONE_TOKEN:
+        from_secret: drone_token
 
 trigger:
   event:
     - promote
     - rollback
   target:
-    - production
\ No newline at end of file
+    - production
+
+---
+depends_on: null
+kind: pipeline
+type: kubernetes
+name: deploy-devcenter-stage-trigger
+
+steps:
+  - name: checkout-to-current-release
+    image: alpine/git
+    environment:
+      APP_RELEASE:
+        from_secret: app_release_staging
+    commands:
+    # "$${APP_RELEASE}" is the latest staging revision that was promoted successfully.
+    - git checkout "$${APP_RELEASE}"
+
+  - name: publish-staging-from-trigger
+    image: plugins/kaniko-ecr
+    environment:
+      NPM_AUTH:
+        from_secret: NPM_AUTH
+      NPM_EMAIL:
+        from_secret: NPM_EMAIL
+      STRAPI_URL:
+        from_secret: STRAPI_URL_STAGING
+      APP_RELEASE:
+        from_secret: app_release_staging
+    settings:
+      create_repository: true
+      enable_cache: true
+      registry: 795250896452.dkr.ecr.us-east-1.amazonaws.com
+      repo: devrel/${DRONE_REPO_NAME}
+      build_args:
+        - NPM_AUTH
+        - NPM_EMAIL
+        - STRAPI_URL
+        - APP_RELEASE
+      tags:
+        - devcenter-staging-build-${DRONE_BUILD_NUMBER}
+      access_key:
+        from_secret: ecr_access_key
+      secret_key:
+        from_secret: ecr_secret_key
+      values_files: ["environments/staging.yaml"]
+
+  - name: deploy-staging-from-trigger
+    image: drone/cli:1.4.0-alpine
+    commands:
+    - drone build promote mongodb/devcenter ${DRONE_BUILD_NUMBER} staging
+    environment:
+      DRONE_SERVER: ${DRONE_SYSTEM_PROTO}://${DRONE_SYSTEM_HOST}
+      DRONE_TOKEN:
+        from_secret: drone_token
+
+trigger:
+  branch:
+    - main
+  event:
+    - custom
```

Comment:
This will only run on `custom` event type, which is what the API will trigger.

Reply:
I'm assuming there will be only one webhook, but I can add additional configuration. Currently, it's only the environment, but I think it makes sense to include another parameter to check.